extensions/amp-story/1.0/amp-story.js

```diff
@@ -990,7 +990,8 @@ export class AmpStory extends AMP.BaseElement {
       this.element.addEventListener(AmpEvents.DOM_UPDATE, ({target}) => {
         this.liveStoryManager_.update(
           target,
-          this.element.querySelectorAll('amp-story-page:not([ad])')
+          this.element.querySelectorAll('amp-story-page:not([ad])'),
+          this.systemLayer_
```

Comment:
Having hard dependencies like this one is what we've been trying to move away from :( The idea is to move over the time to something that's more like what cool kids call "reactive programming"

Can we find another pattern, like maybe having an array of the `segmentIds` in the store?
That might be more refactoring, but if we can remove some other hard dependencies from `amp-story.js` too, that'd be awesome

Reply:
Done.